src/main/java/org/opensearch/knn/plugin/KNNPlugin.java

```diff
@@ -359,6 +374,16 @@ public Settings additionalSettings() {
             IndexModule.INDEX_STORE_HYBRID_MMAP_EXTENSIONS.getDefault(Settings.EMPTY).stream(),
             engineSettings.stream()
         ).collect(Collectors.toList());
-        return Settings.builder().putList(IndexModule.INDEX_STORE_HYBRID_MMAP_EXTENSIONS.getKey(), combinedSettings).build();
+
+        settingsBuilder.putList(IndexModule.INDEX_STORE_HYBRID_MMAP_EXTENSIONS.getKey(), combinedSettings);
+    }
+
+    private Settings pluginAdditionalSettings(final Settings templateAndRequestSettings) {
```

Comment:
nit: can pluginAdditionalSettings be place above additionalMMapFileExtensionsForHybridFs because pluginAdditionalSettings calls  additionalMMapFileExtensionsForHybridFs?

Reply:
ack